hatch/files/setup.py

```diff
@@ -43,7 +43,7 @@
         'Programming Language :: Python :: Implementation :: CPython',{pypy}    ],
     'install_requires': REQUIRES,
     'tests_require': ['coverage', 'pytest'],
-    'packages': find_packages(),{entry_point}
+    'packages': find_packages(exclude=('tests',)),{entry_point}
```

Comment:
This change is needed in case there are subpackages under tests.  Yes it is weird that it is necessary, but it is, and without it there are some very unpredictable results because the top level contents were removed and pytest will do some strange things while collecting the subpackages.

```diff
-exclude=('tests',)
+exclude=('tests', 'tests.*')
```

I also usually add `build` and .* below it, but I've forgotten why I add that.
```py
exclude=('build', 'build.*', 'tests', 'tests.*')
```

Reply:
ping @Kwpolska

Reply:
I deleted my fork, so I can’t make any changes. Besides, this project is basically dead, with this PR not touched for over a year. I’ll close it.

Reply:
@Kwpolska, we had a PR merged two days ago, so there is a bit of traction.
It would be great if you could resubmit an updated PR, or I'll submit one if I dont hear back from you, as I have included your patch in the openSUSE package for this tool.  RedHat also has a maintained package https://repology.org/project/hatch/versions, and possibly Arch also does. https://repology.org/project/python:hatch/versions

Reply:
Feel free to do whatever; this little change is not copyrightable or anything, and frankly, I don’t care.